lib/src/_shared/PickerPopper.tsx

```diff
@@ -0,0 +1,143 @@
+  /**
+   * Custom component for [Transition](https://material-ui.com/components/transitions/#transitioncomponent-prop).
+   */
+  TransitionComponent?: React.ComponentType<TransitionProps>;
+}
+
+export interface PickerPopperProps extends ExportedPickerPopperProps, PaperProps {
+  role: 'tooltip' | 'dialog';
+  TrapFocusProps?: Partial<TrapFocusProps>;
+  anchorEl: PopperProps['anchorEl'];
+  open: PopperProps['open'];
+  onClose: () => void;
+  onOpen: () => void;
+}
+
+export const useStyles = makeStyles(
+  theme => ({
+    popper: {
+      zIndex: theme.zIndex.modal,
+    },
+    paper: {
+      transformOrigin: 'top center',
+      '&:focus': {
+        outline: 'auto',
+        [IS_TOUCH_DEVICE_MEDIA]: {
+          outline: 0,
+        },
+      },
+    },
+    topTransition: {
+      transformOrigin: 'bottom center',
+    },
+  }),
+  { name: 'MuiPickersPopper' }
+);
+
+export const PickerPopper: React.FC<PickerPopperProps> = ({
+  role,
+  PopperProps,
+  TrapFocusProps,
+  TransitionComponent = Grow,
+  open,
+  innerRef = null,
+  anchorEl,
+  children,
+  onClose,
+  onOpen,
+  ...other
+}) => {
+  const classes = useStyles();
+  const lastFocusedElementRef = React.useRef<Element | null>(null);
+  const paperRef = React.useRef<HTMLElement>(null);
+  const handlePopperRef = useForkRef(paperRef, innerRef);
+
+  useGlobalKeyDown(open, {
+    [keycode.Esc]: onClose,
+  });
+
+  React.useEffect(() => {
+    if (role === 'tooltip') {
+      return;
+    }
+
+    if (open) {
+      lastFocusedElementRef.current = document.activeElement;
+    } else if (
+      lastFocusedElementRef.current &&
+      lastFocusedElementRef.current instanceof HTMLElement
+    ) {
+      lastFocusedElementRef.current.focus();
+    }
+  }, [open, role]);
+
+  const handleBlur = () => {
+    if (!open) {
+      return;
+    }
+
+    // document.activeElement is updating on the next tick after `blur` called
+    executeInTheNextEventLoopTick(() => {
+      if (paperRef.current?.contains(document.activeElement)) {
+        return;
+      }
+
+      onClose();
+    });
+  };
+
+  return (
+    <Popper
+      transition
+      role={role}
+      open={open}
+      anchorEl={anchorEl}
+      placement="bottom"
+      className={clsx(classes.popper, PopperProps?.className)}
+      {...PopperProps}
+    >
+      {({ TransitionProps, placement }) => (
+        <TrapFocus
+          open={open}
+          disableAutoFocus
+          disableEnforceFocus={role === 'tooltip'}
+          isEnabled={() => true}
+          getDoc={() => paperRef.current?.ownerDocument ?? document}
+        >
+          <TransitionComponent {...TransitionProps} onEntered={onOpen}>
+            <Paper
+              tabIndex={-1}
+              elevation={8}
+              ref={handlePopperRef}
+              className={clsx(classes.paper, {
+                [classes.topTransition]: placement === 'top',
+              })}
+              onBlur={handleBlur}
+              {...other}
```

Comment:
`other` should be spread on the root element: Popper.

Reply:
Private component or not, I think that they should all follow this rule. I believe we already enforce it.